src/test/java/org/mars/rover/kata/KataTest.java

```diff
@@ -0,0 +1,99 @@
+package org.mars.rover.kata;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+import java.util.Scanner;
+
+public class KataTest {
+    @Test
+    public void testRoverClassExists() {
+        String className = "org.mars.rover.kata.MarsRover";
+
+        try {
+            Class<?> myClass = Class.forName(className);
+            Assertions.assertNotNull(myClass);
+            Assertions.assertEquals(className, myClass.getName());
+            System.out.println("Class " + className + " exists.");
+        } catch (ClassNotFoundException e) {
+            Assertions.fail("Class " + className + " does not exist.");
+        }
+    }
+
+    @Test
+    public void testRoverReceivesStartingPointAndPosition() {
+        MarsRover marsRover = new MarsRover();
+
+        marsRover.setDirection(Direction.N);
+        marsRover.setCoordinate(new Coordinate(0, 1));
+
+        Assertions.assertEquals(marsRover.getDirection(), Direction.N);
+        Assertions.assertEquals(marsRover.getCoordinate().getX(), 0);
+        Assertions.assertEquals(marsRover.getCoordinate().getY(), 1);
+    }
+
+    @Test
+    public void testNavigatorClassExists() {
+        String className = "org.mars.rover.kata.MarsNavigator";
+
+        try {
+            Class<?> myClass = Class.forName(className);
+            Assertions.assertNotNull(myClass);
+            Assertions.assertEquals(className, myClass.getName());
+            System.out.println("Class " + className + " exists.");
+        } catch (ClassNotFoundException e) {
+            Assertions.fail("Class " + className + " does not exist.");
+        }
+    }
+
+    @Test
+    public void testNavigatorConstructsUpperRightCoordinates() {
+        String coordinatesString = "5 5";
+        System.setIn(new ByteArrayInputStream(coordinatesString.getBytes()));
+        MarsNavigator marsNavigator = new MarsNavigator(new Scanner(System.in));
+
+        marsNavigator.loadInput();
+        marsNavigator.processInput();
+        Coordinate coordinate = marsNavigator.getCoordinate(1, 3);
+
+        Assertions.assertEquals(coordinate.getX(), 1);
+        Assertions.assertEquals(coordinate.getY(), 3);
+    }
+
+    @Test
+    public void testNavigatorCreatesRoverInstances() {
```

Comment:
Bardzo fajny pomysł na taki end-to-end test. Konkretne dane wejściowe powinny dać konkretne dane wyjściowe. Takie całościowe sprawdzenie wszystkich mechanizmów. Takie balck-box testing.